### PR TITLE
 Changelog entry for Go MADV; doc cleanup

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -43,6 +43,14 @@ We use *breaking* word for marking changes that are not backward compatible (rel
 This release also disables gossip mode by default for all components.
 See [this](docs/proposals/approved/201809_gossip-removal.md) for more details.
 
+:warning: This release moves Thanos docker images and artifacts to Golang 1.12. This release include change in GC's memory release which gives following effect (source: https://golang.org/doc/go1.12):
```
Comment (GiedriusS) [inline CHANGELOG.md]: `includes a change` ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -43,6 +43,14 @@ We use *breaking* word for marking changes that are not backward compatible (rel
...
+
+> On Linux, the runtime now uses MADV_FREE to release unused memory. This is more efficient but may result in higher reported RSS. The kernel will reclaim the unused data when it is needed. To revert to the Go 1.11 behavior (MADV_DONTNEED), set the environment variable GODEBUG=madvdontneed=1.
+
+To ensure correct memory reporting:
+* Use memory heap allocation metric exposed by Golang.
```
Comment (GiedriusS) [inline CHANGELOG.md]: Maybe it would be worth mentioning that it is `go_memstats_heap_alloc_bytes`

Comment (GiedriusS) [reply CHANGELOG.md]: And maybe `use` -> `look at` or `observe`

File: CHANGELOG.md [hunk 3]
```diff
@@ -43,6 +43,14 @@ We use *breaking* word for marking changes that are not backward compatible (rel
 This release also disables gossip mode by default for all components.
 See [this](docs/proposals/approved/201809_gossip-removal.md) for more details.
 
+:warning: This release moves Thanos docker images and artifacts to Golang 1.12. This release include change in GC's memory release which gives following effect (source: https://golang.org/doc/go1.12):
+
+> On Linux, the runtime now uses MADV_FREE to release unused memory. This is more efficient but may result in higher reported RSS. The kernel will reclaim the unused data when it is needed. To revert to the Go 1.11 behavior (MADV_DONTNEED), set the environment variable GODEBUG=madvdontneed=1.
+
+To ensure correct memory reporting:
```
Comment (GiedriusS) [inline CHANGELOG.md]: Is it really incorrect with those changes? Maybe `precise` would fit here more

Comment (GiedriusS) [inline CHANGELOG.md]: Is it really incorrect with those changes? Maybe `precise` would fit here more

Comment (bwplotka) [reply CHANGELOG.md]: true

File: CHANGELOG.md (hunk 2 above)
Comment (bwplotka) [reply CHANGELOG.md]: Thanks Man I need your memory (:

**Comment by bwplotka** [comment]
PTAL @GiedriusS and thanks for review

File: CHANGELOG.md (hunk 1 above)
Comment (povilasv) [reply CHANGELOG.md]: We've been using Go 1.12 for awhile now, not in this release https://github.com/improbable-eng/thanos/pull/868

This release changes to 1.12.5, please change the text to note that and add link to https://github.com/improbable-eng/thanos/pull/1156

File: cmd/thanos/sidecar.go [hunk 4]
```diff
@@ -305,29 +305,26 @@ func runSidecar(
 }
 
 func validatePrometheus(ctx context.Context, logger log.Logger, m *promMetadata) error {
-	flags := promclient.Flags{
-		TSDBMinTime: model.Duration(2 * time.Hour),
-		TSDBMaxTime: model.Duration(2 * time.Hour),
-	}
+	var (
```
Comment (povilasv) [inline cmd/thanos/sidecar.go]: I don't think this code change should be part of this PR, or am I missing smth?

**Comment by povilasv** [comment]
Also We've been using Go 1.12 for awhile now, not in this release #868

This release changes to 1.12.5, please change the text to note that and add link to #1156

File: cmd/thanos/sidecar.go (hunk 4 above)
Comment (bwplotka) [reply cmd/thanos/sidecar.go]: It is intentionally there, but can split if needed

**Comment by bwplotka** [comment]
I simplified PR with just doc and CHANGELOG change. 

@povilasv please read carefully.
 
> Also We've been using Go 1.12 for awhile now, not in this release #868

Yes that's why I updated 0.4.0 release.

> This release changes to 1.12.5, please change the text to note that and add link to #1156

It's already there for newest unlreleased.

**Comment by povilasv** [comment]
:+1:

**Pull request merged**